src/controllers/trainee/Controller.ts

```diff
@@ -0,0 +1,112 @@
+import { Request, Response } from 'express';
+import { UserRepository } from '../../repositories';
+
+const userRepository = new UserRepository();
+
+class TraineeRoutes {
+// Function to fetch all trainees
+    public async get(request: Request, response: Response, next) {
+        try {
+            const {skip , limit} = request.query;
+            const count = await userRepository.count();
+            const result = await userRepository.getAll({ role: 'trainee' }, {password: 0}, { skip, limit});
+            response.send({
+                data: {
+                    count,
+                    records: result,
+                },
+                message: 'Successfully fetched trainees',
+                status: 'OK',
+            });
+        }
+        catch (err) {
+            next({
+                error: 'Bad Request',
+                message: err.message || 'Could not fetch data',
+                status: 400,
+            });
+        }
+
+    }
+
+// function to create new trainee
+    public async create(request: Request, response: Response, next) {
+        try {
+            const { email, password, name } = request.body;
+            const data = {
+                email,
+                name,
+                password,
+            };
+            const result = await userRepository.createWithHash(data, request.user._id);
+            response.send({
+                data: result,
+                message: 'Trainee Created Successfully',
+                status: 'ok',
+            });
+        }
+        catch (err) {
+            next({
+                error: 'Bad Request',
+                message: err.message,
+                status: 400,
+            });
+        }
+    }
+
+// function to update trainee
+    public async update(request: Request, response: Response, next) {
+        try {
+            const allowed = ['name', 'email', 'password'];
+            const data = Object.keys(request.body.dataToUpdate)
```

Comment:
Please use destructuring for dataToUpdate.

Reply:
Ok, thanks
Done